### PR TITLE
Form save split button

### PR DESCRIPTION
**Pull request by manuelblum**
- [x] FromSaveSplitButton
- [x] Changelog
- [x] dokumentation

Screen Recording 2021-06-02 at 16 38 56 mov

### CODE REVIEW

**Review by dkarnutsch — APPROVED** [review]

File: packages/admin/src/FormSaveSplitButton.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                    form.submit();
+                }}
+            >
+                <FormattedMessage id="comet.generic.save" defaultMessage="Save" />
+            </SaveButton>
+            <SaveButton
+                color={"primary"}
+                variant={"contained"}
+                saving={saving}
+                hasErrors={hasErrors != null}
```
Comment (nsams) [inline packages/admin/src/FormSaveSplitButton.tsx:39]: Form state does contain submitting and hasSubmitErrors. We could use that and avoid having to pass those as props. 

As this component can't be used without FinalForm you could even completely rely on those and remove the saving/hasErrors props.

File: packages/admin/src/FormSaveSplitButton.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { SplitButton } from "./common/buttons/split/SplitButton";
+import { useStackApi } from "./stack/Api";
+
+export interface FormSaveButtonProps {
+    localStorageKey?: string;
+    saving: boolean;
+    hasErrors: boolean;
+}
+
+export const FormSaveSplitButton = ({ saving, hasErrors, localStorageKey }: PropsWithChildren<FormSaveButtonProps>) => {
```
Comment (johnnyomair) [inline packages/admin/src/FormSaveSplitButton.tsx:17]: Should we rename this component to `FinalFormSaveSplitButton`? Doing so would be more in line with our other `final-form` related components.

File: packages/admin/src/FinalFormSaveSplitButton.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                <FormattedMessage id="comet.generic.save" defaultMessage="Save" />
+            </SaveButton>
+            <SaveButton
+                color={"primary"}
+                variant={"contained"}
+                saving={submitting}
+                hasErrors={hasSubmitErrors}
+                onClick={async () => {
+                    const submitResult = await form.submit();
+                    const error = submitResult?.[FORM_ERROR];
```
Comment (johnnyomair) [inline packages/admin/src/FinalFormSaveSplitButton.tsx:40]: A `final-form` can also return errors for individual fields using `{ [fieldName]: "Error" }`. Therefore, the form is only valid if `submitResult` is `undefined`. Your check would not work for field-level submission errors

Comment (nsams) [reply packages/admin/src/FinalFormSaveSplitButton.tsx:40]: Totally missed that

**Review by nsams — APPROVED** [review]

**Review by kaufmo — APPROVED** [review]

File: packages/admin/src/FinalFormSaveSplitButton.tsx (hunk 3 above)
Comment (manuelblum) [reply packages/admin/src/FinalFormSaveSplitButton.tsx:40]: fixed that

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**